Flaky test: test_postgresql_replica_database_engine/test_1.py::test_abrupt_connection_loss_while_heavy_replication

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_postgresql_replica_database_engine/test_1.py::test_abrupt_connection_loss_while_heavy_replication
Failure reason: cannot pause container
CI report: [Integration tests (arm_binary, distributed plan, 2/4)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=93360&sha=8f72452d7099a4dfb6d5deb3f3a31ccf48c9745c&name_0=PR&name_1=Integration%20tests%20%28arm_binary%2C%20distributed%20plan%2C%202%2F4%29)

CIDB statistics: [cidb](https://play.clickhouse.com/play?user=play&run=1#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X3Bvc3RncmVzcWxfcmVwbGljYV9kYXRhYmFzZV9lbmdpbmUvdGVzdF8xLnB5Ojp0ZXN0X2FicnVwdF9jb25uZWN0aW9uX2xvc3Nfd2hpbGVfaGVhdnlfcmVwbGljYXRpb24nCiAgICAtLSBBTkQgY2hlY2tfbmFtZSA9ICdJbnRlZ3JhdGlvbiB0ZXN0cyAoYXJtX2JpbmFyeSwgZGlzdHJpYnV0ZWQgcGxhbiwgMi80KScKICAgIEFORCB0ZXN0X3N0YXR1cyBJTiAoJ0ZBSUwnLCAnRVJST1InKQogICAgQU5EICgocHVsbF9yZXF1ZXN0X251bWJlciA9IDAgQU5EIGhlYWRfcmVmID0gJ21hY29zLWplbWFsbG9jLWRpcnR5LWRlY2F5JykgT1IgKHB1bGxfcmVxdWVzdF9udW1iZXIgIT0gMCBBTkQgYmFzZV9yZWYgPSAnbWFzdGVyJykpCkdST1VQIEJZIGRheQpPUkRFUiBCWSBkYXkgREVTQwo=)

Test output:
```
File: test_postgresql_replica_database_engine/test_1.py:310 - in test_abrupt_connection_loss_while_heavy_replication
    with started_cluster.pause_container("postgres1"):
File: /usr/lib/python3.10/contextlib.py:135 - in __enter__
    return next(self.gen)
File: helpers/cluster.py:4034 - in pause_container
    self._pause_container(instance_name)
File: helpers/cluster.py:4023 - in _pause_container
    subprocess_check_call(self.base_cmd + ["pause", instance_name])
File: helpers/cluster.py:294 - in subprocess_check_call
    return run_and_check(args, detach=detach, nothrow=nothrow, **kwargs)
File: helpers/cluster.py:191 - in run_and_check
    raise Exception(
E   Exception: Command [docker compose --env-file /home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests/integration/test_postgresql_replica_database_engine/_instances-1-gw3/.env --project-name roottestpostgresqlreplicadatabaseengine1-gw3 --file /home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests/integration/test_postgresql_replica_database_engine/_instances-1-gw3/instance/docker-compose.yml --file /home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests/integration/helpers/../../../tests/integration/compose/docker_compose_postgres.yml pause postgres1] return non-zero code 1: Error response from daemon: cannot pause container 71afd8f5158af39024bc34d6ed15ae4eef73a72c73a743397f735635397e18ae: OCI runtime pause failed: unable to freeze: unknown
```

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -4032,6 +4032,12 @@ class ClickHouseCluster:
     def _unpause_container(self, instance_name):
         subprocess_check_call(self.base_cmd + ["unpause", instance_name])
 
+    def _pause_container_using_signal(self, instance_name):
+        subprocess_check_call(self.base_cmd + ["kill", "--signal=SIGSTOP", instance_name])
+
+    def _unpause_container_using_signal(self, instance_name):
+        subprocess_check_call(self.base_cmd + ["kill", "--signal=SIGCONT", instance_name])
+
     @contextmanager
     def pause_container(self, instance_name):
         """Use it as following:
@@ -4043,6 +4049,14 @@ class ClickHouseCluster:
             yield
         finally:
             self._unpause_container(instance_name)
+
+    @contextmanager
+    def pause_container_using_signal(self, instance_name):
+        self._pause_container_using_signal(instance_name)
+        try:
+            yield
+        finally:
+            self._unpause_container_using_signal(instance_name)
 
     def open_bash_shell(self, instance_name):
         os.system(" ".join(self.base_cmd + ["exec", instance_name, "/bin/bash"]))

--- a/tests/integration/test_postgresql_replica_database_engine/test_1.py
+++ b/tests/integration/test_postgresql_replica_database_engine/test_1.py
@@ -307,7 +307,7 @@ def test_abrupt_connection_loss_while_heavy_replication(started_cluster):
     time.sleep(2)
 
 
-    with started_cluster.pause_container("postgres1"):
+    with started_cluster.pause_container_using_signal("postgres1"):
         # for i in range(NUM_TABLES):
         #     result = instance.query(f"SELECT count() FROM test_database.postgresql_replica_{i}")
         #     print(result) # Just debug

